### DISCUSSION TITLE
“test_y" should not be allowed to be accessed！！！

### DISCUSSION THREAD

**Discussion by gengdd** [PaddlePaddle/PaddleSpatial#74] [BaiduKDDCup2022] [answered]
According to the official code, we can know that the test_x and test_y folders of the online test dataset are in the same directory. This way we can easily read the ground truth  in test_y and return it as the predicted value. Therefore, I suggest to place the ground truth in test_y in other files of the background server, and the name of the path cannot use "path_to_test_y" as the key in the settings dictionary, “path_to_test_y“ is used to ensure that our offline evaluation can run through, online evaluation should use a secret key, or do not save the location information of test_y in the settings dictionary. In short, access to the ground truth of the test dataset needs to be disabled！！！

**Comment by gengdd** [comment]
If possible, the rank list should be cleared and resubmitted! ! !

**Comment by ramber1836** [comment, marked as answer]
Hi, there,

The released evaluation code is for offline test and demonstration.
The real evaluation script adopted for online test does not locate the
tast_x and test_y in the same folder.
Besides, the location of the main evaluation script is invisible for the
submitted code and the submitted code is extracted in a temporary path.
Above all, we can ensure the security of most cases.

Some false high scores have been fixed, it is still in progress.

In addition, a hundred percent security is not easy, it would be
very welcomed if you can provide us the clues to any potential risks.

Thanks!

Regards,
Xinjiang Lu
